craft_application/commands/lint.py

```diff
@@ -0,0 +1,162 @@
+    overview = "Run linters against the project (pre) or artifacts (post)."
+
+    def fill_parser(self, parser: argparse.ArgumentParser) -> None:
+        """Add command-line arguments for lint configuration."""
+        parser.add_argument(
+            "--stage",
+            choices=[Stage.PRE.value, Stage.POST.value],
+            default=Stage.PRE.value,
+            help="When to lint: 'pre' = source tree, 'post' = built artifacts.",
+        )
+        parser.add_argument(
+            "--lint-ignore",
+            action="append",
+            dest="lint_ignores",
+            type=_parse_ignore_rule,
+            default=[],
+            metavar="RULE",
+            help=(
+                "Ignore rule of the form 'linter:id' (ignore id everywhere) or "
+                "'linter:id=glob' (ignore id when filename matches glob). May repeat."
+            ),
+        )
+        parser.add_argument(
+            "--lint-ignore-file",
+            action="append",
+            type=Path,
+            dest="lint_ignore_files",
+            default=[],
+            metavar="PATH",
+            help=("Path to YAML ignore file. May repeat. CLI rules take precedence."),
+        )
+
+    def run(self, parsed_args: argparse.Namespace) -> int | None:
+        """Execute lint for the requested stage and return the exit code."""
+        services = self._services
+        stage = Stage(parsed_args.stage)
+
+        # Resolve project dir
+        project_service = services.get("project")
+        project_dir: Path = project_service.resolve_project_file_path().parent
+        if stage == Stage.POST and not project_service.is_configured:
+            project_service.configure(platform=None, build_for=None)
+            project_service.get()
+
+        artifact_dirs: list[Path] = []
+        if stage == Stage.POST:
+            lifecycle = services.get("lifecycle")
+            artifact_dirs = [lifecycle.prime_dir]
+
+        ctx = LintContext(project_dir=project_dir, artifact_dirs=artifact_dirs)
+
+        linter = services.get("linter")
+        cli_ignore_rules = list(parsed_args.lint_ignores or [])
+        cli_ignore_config = _build_cli_ignore_config(cli_ignore_rules)
+        linter.load_ignore_config(
+            project_dir=project_dir,
+            cli_ignores=cli_ignore_config or None,
+            cli_ignore_files=list(parsed_args.lint_ignore_files or []),
+        )
+        issues = list(linter.run(stage, ctx))
+
+        if issues:
+            emit.message("lint results:")
+            for linter_name, linter_issues in linter.issues_by_linter.items():
+                emit.message(f"{linter_name}:")
+                for issue in linter_issues:
+                    location = f" ({issue.filename})" if issue.filename else ""
+                    emit.message(
+                        f"  - [{issue.severity.name}] {issue.id}: {issue.message}{location}"
+                    )
```

Comment:
From some initial testing, it seems that YAML syntax errors never reach this code. Instead, `YamlError` (I believe)  interrupts. Is this expected?

Source:

```yaml
name: testcraft-test
version: "1.0"

base: ubuntu@24.04
platforms:
  platform-independent:
    build-on: [amd64, arm64, ppc64el, s390x, riscv64]
    build-for: [all]

parts:
  my-test:
    plugin: nil

parts:
```

Result:

```bash
$ testcraft lint
error parsing '(unknown)': found duplicate key 'parts'
Detailed information: 
while constructing a mapping
found duplicate key 'parts'
  in "<unicode string>", line 1, column 1:
    name: testcraft-test
    ^
Recommended resolution: Ensure (unknown) contains valid YAML
```